cg/meta/archive/ddn_dataflow.py

```diff
@@ -188,23 +192,29 @@ def auth_header(self) -> Dict[str, str]:
             self._refresh_auth_token()
         return {"Authorization": f"Bearer {self.auth_token}"}
 
-    def archive_folders(self, sources_and_destinations: List[MiriaFile]) -> int:
+    def archive_folders(self, files_and_samples: List[FileAndSample]) -> int:
```

Comment:
Would this only work for MiriaFiles or can this be easily extended to other File types? From the type hits it looks like it

Reply:
@ChrOertlin Ideally I would like to rename the whole class to miria_dataflow. I think the naming "DDNDataFlowApi" is an artifact from before it was decided that Miria was more descriptive. So ArchiveHandler has the general method "archive_folders" whereas DDNDataFlowApi has an implementation which is only intended to be used for Karolinska customers. 

The answer to your question would depend on what you mean by "work". If you provide a list of FileAndSamples, the implementation in DDNDataFlowApi will convert these to a list of MiriaFiles and send a request to store these using Miria. So it would probably execute regardless of what files and samples are provided, but it is only intended for storage of files via Miria.

Any other implementation of archive_folders would convert the files and samples to the format used for archiving there. Say FileAndSample -> OtherArchivingApiFile -> archive using OtherArchivingApi.

Reply:
Sorry I meant to point to this part of the code. Why would we care about miriafiles here? can it not be a general file class that determines internally if it are miria files or something else?

```Python 
   miria_file_data: List[MiriaFile] = self.convert_into_transfer_data(
            files_and_samples, is_archiving=True
```